<!DOCTYPE html>
<html lang="zh-Hans">
<head>

  <meta charset="utf-8" />

  
  <title>病毒整合位点分析软件CTAT-vif的安装排雷指南</title>

  
  
  <link rel="stylesheet"
        href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.4.0/styles/base16/dracula.min.css">
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.4.0/languages/python.min.js"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.4.0/languages/r.min.js"></script>
  <script>hljs.highlightAll();</script>

  
  <link href="//cdn.jsdelivr.net" rel="dns-prefetch">
  <link href="//cdnjs.cloudflare.com" rel="dns-prefetch">
  
  <link href="//use.fontawesome.com" rel="dns-prefetch">
  
  <link href="//fonts.googleapis.com" rel="dns-prefetch">
  <link href="//fonts.gstatic.com" rel="dns-prefetch">
  <link href="///disqus.com" rel="dns-prefetch">
  <link href="//c.disquscdn.com" rel="dns-prefetch">
  
  <link href="//www.google-analytics.com" rel="dns-prefetch">
  

  

  
  <meta name="author" content="Wang Jiaxuan">
  <meta name="description" content="这个软件很强大，但是坑也特别多，但是至少也比那些常年不更新，连给bug的机会都不给，因为你根本安装不上的的整合分析软件强多了！！！
">

  
  
    <meta name="twitter:card" content="summary_large_image">
    <meta name="twitter:site" content="@abego13">
    <meta name="twitter:title" content="病毒整合位点分析软件CTAT-vif的安装排雷指南">
    <meta name="twitter:description" content="这个软件很强大，但是坑也特别多，但是至少也比那些常年不更新，连给bug的机会都不给，因为你根本安装不上的的整合分析软件强多了！！！
">
    <meta name="twitter:image" content="https://abego.cn/media/cover.jpg">
  

  
  <meta property="og:type" content="article">
  <meta property="og:title" content="病毒整合位点分析软件CTAT-vif的安装排雷指南">
  <meta property="og:description" content="这个软件很强大，但是坑也特别多，但是至少也比那些常年不更新，连给bug的机会都不给，因为你根本安装不上的的整合分析软件强多了！！！
">
  <meta property="og:url" content="https://abego.cn/2022/01/16/ctat-vif-find-the-virus-integration-site-in-human-genome/">
  <meta property="og:image" content="https://abego.cn/media/cover.jpg">




<meta name="generator" content="Hugo 0.91.2">


<link rel="canonical" href="https://abego.cn/2022/01/16/ctat-vif-find-the-virus-integration-site-in-human-genome/">

<meta name="renderer" content="webkit">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="format-detection" content="telephone=no,email=no,adress=no">
<meta http-equiv="Cache-Control" content="no-transform">


<meta name="robots" content="index,follow">
<meta name="referrer" content="origin-when-cross-origin">







<meta name="theme-color" content="#02b875">
<meta name="apple-mobile-web-app-capable" content="yes">
<meta name="apple-mobile-web-app-status-bar-style" content="black">
<meta name="apple-mobile-web-app-title" content="Woney&#39;s Blog">
<meta name="msapplication-tooltip" content="Woney&#39;s Blog">
<meta name='msapplication-navbutton-color' content="#02b875">
<meta name="msapplication-TileColor" content="#02b875">
<meta name="msapplication-TileImage" content="/icons/icon-144x144.png">
<link rel="icon" href="https://abego.cn/favicon.ico">
<link rel="icon" type="image/png" sizes="16x16" href="https://abego.cn/icons/icon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="https://abego.cn/icons/icon-32x32.png">
<link rel="icon" sizes="192x192" href="https://abego.cn/icons/icon-192x192.png">
<link rel="apple-touch-icon" href="https://abego.cn/icons/icon-152x152.png">
<link rel="manifest" href="https://abego.cn/manifest.json">


<link rel="preload" href="https://abego.cn/styles/main-rendered.min.css" as="style">


<link rel="preload" href="https://fonts.googleapis.com/css?family=Lobster" as="style">
<link rel="preload" href="https://abego.cn/images/avatar.png" as="image">
<link rel="preload" href="https://abego.cn/images/grey-prism.svg" as="image">


<style>
  body {
    background: rgb(244, 243, 241) url('/images/grey-prism.svg') repeat fixed;
  }
</style>
<link rel="stylesheet" href="https://abego.cn/styles/main-rendered.min.css">

<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.1/css/all.css" ZgotmplZ crossorigin="anonymous">


<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lobster">



<script src="https://cdn.jsdelivr.net/npm/medium-zoom@1.0.2/dist/medium-zoom.min.js"></script>





<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/video.js@7.3.0/dist/video-js.min.css">



  
  
<!--[if lte IE 8]>
  <script src="https://cdn.jsdelivr.net/npm/html5shiv@3.7.3/dist/html5shiv.min.js"></script>
  <script src="https://cdn.jsdelivr.net/npm/respond.js@1.4.2/dest/respond.min.js"></script>
  <script src="https://cdn.jsdelivr.net/npm/videojs-ie8@1.1.2/dist/videojs-ie8.min.js"></script>
<![endif]-->

<!--[if lte IE 9]>
  <script src="https://cdn.jsdelivr.net/npm/eligrey-classlist-js-polyfill@1.2.20180112/classList.min.js"></script>
<![endif]-->


</head>
  <body>
    <div class="suspension">
      <a role="button" aria-label="Go to top" title="Go to top" class="to-top is-hide"><span class="fas fa-caret-up" aria-hidden="true"></span></a>
      
        
	<a role="button" aria-label="Go to comments" title="Go to comments" class="to-comment" href="#disqus_thread"><span class="fas fa-comment-dots" aria-hidden="true"></span></a>
        
      
    </div>
    
    
  <header class="site-header">
  <a href="https://abego.cn/"><img class="avatar" src="https://abego.cn/images/avatar.png" alt="Avatar"></a>
  
  <h2 class="title"><a href="https://abego.cn/">Woney&#39;s Blog</a></h2>
  
  <p class="subtitle">生活 学习 工作</p>
  <button class="menu-toggle" type="button" aria-label="Main Menu" aria-expanded="false" tab-index="0">
    <span><i class="fas fa-bars"></i></span>
  </button>
  <nav class="site-menu collapsed">
    <h2 class="offscreen">Main Menu</h2>
    <ul class="menu-list">
      
      
      
      
        <li class="menu-item
          
          
          ">
          <a href="https://abego.cn/tags/">查找</a>
        </li>
      
        <li class="menu-item
          
          
           is-active">
          <a href="https://abego.cn/">主页</a>
        </li>
      
        <li class="menu-item
          
          
          ">
          <a href="https://abego.cn/resume/">简历</a>
        </li>
      
        <li class="menu-item
          
          
          ">
          <a href="https://abego.cn/links/">Links</a>
        </li>
      
        <li class="menu-item
          
          
          ">
          <a href="https://github.com/wangjiaxuan666">代码</a>
        </li>
      
        <li class="menu-item
          
          
          ">
          <a href="https://abego.cn/about/">吐槽</a>
        </li>
      
    </ul>
  </nav>
  <nav class="social-menu collapsed">
    <h2 class="offscreen">Social Networks</h2>
    <ul class="social-list"><li class="social-item">
          <a href="mailto:992914078@qq.com" title="Email" aria-label="Email">
            <span class="fas fa-envelope" aria-hidden="true"></span>
          </a>
        </li><li class="social-item">
          <a href="//github.com/wangjiaxuan666" rel="me" title="GitHub" aria-label="GitHub">
      <span class="fab fa-github" aria-hidden="true"></span>
          </a>
        </li><li class="social-item">
          <a href="//twitter.com/abego13" rel="me" title="Twitter" aria-label="Twitter">
            <span class="fab fa-twitter" aria-hidden="true"></span>
          </a>
        </li><li class="social-item">
          <a href="https://abego.cn/images/qrcode.jpg" rel="me" title="Wechat" aria-label="Wechat">
            <span class="fab fa-weixin" aria-hidden="true"></span>
          </a>
        </li></ul>
  </nav>
</header>

  <section class="main post-detail">
    <header class="post-header">
      <h1 class="post-title">病毒整合位点分析软件CTAT-vif的安装排雷指南</h1>
      <p class="post-meta">@Wang Jiaxuan · Jan 16, 2022 · 4 min read</p>
    </header>
    
    <article class="post-content"><p>这个软件很强大，但是坑也特别多，但是至少也比那些常年不更新，连给bug的机会都不给，因为你根本安装不上的的整合分析软件强多了！！！</p>
<h1 id="基因整合位点的检测">基因整合位点的检测</h1>
<h2 id="01版本安装">0.1版本安装</h2>
<p>用的软件是放在Github上的<a href="https://github.com/broadinstitute/CTAT-VirusIntegrationFinder">CTAT-VirusIntegrationFinder</a>软件，这个软件是麻省理工学院和哈佛大学联合Broad Institute开发的软件，这个研究所声名远扬，如雷贯耳，去他们的Github主页看，就知道是开发picard和GTAK的大牛研究机构。</p>
<p><a href="https://github.com/broadinstitute/CTAT-VirusIntegrationFinder">CTAT-VirusIntegrationFinder</a>是<a href="https://github.com/NCIP/Trinity_CTAT/wiki">Trinity Cancer Transcriptome Analysis Toolkit</a> 工具套件其中一个模块，CTAT-VirusIntegrationFinder顾名思义，主要是做病毒基因整合的检测。而整个套件中其他模块有做融合基因，有作突变位点，也有做CNV和SV的。总之就是很强大。</p>
<h3 id="安装">安装</h3>
<p>现在<a href="https://github.com/broadinstitute/CTAT-VirusIntegrationFinder/releases">Releases · broadinstitute/CTAT-VirusIntegrationFinder (github.com)</a>上下载软件包，然后解压缩，看到一个download的sh文件，bash运行，注意要在联网环境下</p>
<div class="highlight"><pre tabindex="0" style="background-color:#fff;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-bash" data-lang="bash">tar -zxvf CTAT-VirusIntegrationFinder-v1.0.1.FULL.tar.gz
bash  download_cromwell.sh
</code></pre></div><p>注意，cromwell 必须要jdk 1.8.0+ 的环境，具体可以看我当时提的<a href="https://github.com/broadinstitute/CTAT-VirusIntegrationFinder/issues/41">issue</a>.这第一个坑，卡我了至少三天。</p>
<p>下载好后，就可以用<code>make</code>命令编译，如果没有报错，那么恭喜你你已经跳过安装最大的坑了，如果报错了，需要根据报错信息，进行环境的补全，可能文章后面的0.04版本的安装会给一些提示。</p>
<h3 id="基因组索引">基因组索引</h3>
<p>基因组索引需要在[网站](<a href="https://data.broadinstitute.org/Trinity/CTAT_RESOURCE_LIB/">Index of /Trinity/CTAT_RESOURCE_LIB (broadinstitute.org)</a>)下载基本的源文件, 虽然解压缩后，这个文件夹里有很多文件，但是还是需要重新构建（你敢信，这个又卡了我三天，我以为直接可以用！！！，脸上笑眯眯，心里MMP）</p>
<p>从<a href="https://github.com/NCIP/ctat-genome-lib-builder/">Github</a>上下载构建索引的脚本:</p>
<div class="highlight"><pre tabindex="0" style="background-color:#fff;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-bash" data-lang="bash">git clone https://github.com/NCIP/ctat-genome-lib-builder/
ls ctat-genome-lib-builder/prep_genome_lib.pl
<span style="color:#888"># 查看位置</span>
ctat-genome-lib-builder/prep_genome_lib.pl <span style="color:#04d;background-color:#fff0f0">\
</span><span style="color:#04d;background-color:#fff0f0"></span>--genome_fa GRCh38.primary_assembly.genome.fa <span style="color:#04d;background-color:#fff0f0">\
</span><span style="color:#04d;background-color:#fff0f0"></span>--gtf gencode.v37.annotation.gtf <span style="color:#04d;background-color:#fff0f0">\
</span><span style="color:#04d;background-color:#fff0f0"></span>--dfam_db human <span style="color:#04d;background-color:#fff0f0">\
</span><span style="color:#04d;background-color:#fff0f0"></span>--fusion_annot_lib fusion_lib.Mar2021.dat.gz <span style="color:#04d;background-color:#fff0f0">\
</span><span style="color:#04d;background-color:#fff0f0"></span>--human_gencode_filter <span style="color:#04d;background-color:#fff0f0">\
</span><span style="color:#04d;background-color:#fff0f0"></span>--pfam_db current
</code></pre></div><p>当这个运行可以构建基因，但是我的环境好像少了perl环境，服气了。</p>
<blockquote>
<p>我还是直接从网站down构建好的基因组吧,果然下载构建好的基因，运行直接成功了。虽然这个构建好的基因组index有31G的大小，贼难下载的说。</p>
</blockquote>
<div class="highlight"><pre tabindex="0" style="background-color:#fff;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-bash" data-lang="bash">~/wangjiaxuan/biosoft/CTAT-vif/ctat-vif <span style="color:#04d;background-color:#fff0f0">\
</span><span style="color:#04d;background-color:#fff0f0"></span>--left ./genome_lib/test.fq   <span style="color:#04d;background-color:#fff0f0">\
</span><span style="color:#04d;background-color:#fff0f0"></span>--genome_lib_dir ~/CTAT-vif/genome_lib/ctat_genome_lib_build_dir <span style="color:#04d;background-color:#fff0f0">\
</span><span style="color:#04d;background-color:#fff0f0"></span>--sample_id yesimola <span style="color:#04d;background-color:#fff0f0">\
</span><span style="color:#04d;background-color:#fff0f0"></span>-O ./test <span style="color:#04d;background-color:#fff0f0">\
</span><span style="color:#04d;background-color:#fff0f0"></span>--cpu <span style="color:#00d;font-weight:bold">2</span> <span style="color:#04d;background-color:#fff0f0">\
</span><span style="color:#04d;background-color:#fff0f0"></span>--virus_type hpv33 <span style="color:#888"># 这个参数是我自己加上的，原软件是没有的</span>
</code></pre></div><p>以上就会运行成功，结果预览：</p>
<blockquote>
<p>注意：因为脚本会默认用的是<code>,</code>来分割read，所以如果你的fq文件的reads名字增加了一些特殊字符是回报错的。所以针对这点需要在此基础上修改脚本，比如我就是修改成以<code>~</code>来作为分割符。</p>
</blockquote>
<h2 id="004版本安装时候的坑">0.04版本安装时候的坑</h2>
<p>最早期的时候安装过这个软件，但是踩了很多坑，后面作者更新（因为我们和作者联系，希望增加SE测序的支持，正好作者准备更新），所以我就重新走了一遍安装和索引流程，但我觉得之前的坑排雷经验也很重要，留下来做个备份。</p>
<p><strong>也许对于其他环境的问题，之前的坑还在，只不过我这次更新，已经把之前的坑填了而已。</strong></p>
<p>言归正传、<a href="https://github.com/broadinstitute/CTAT-VirusIntegrationFinder">CTAT-VirusIntegrationFinder</a>需要在Github上下载，下载（先别下载有坑！看后面），运行参数很简单：</p>
<div class="highlight"><pre tabindex="0" style="background-color:#fff;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-bash" data-lang="bash">tat-VIF.py --left_fq reads_1.fastq <span style="color:#04d;background-color:#fff0f0">\
</span><span style="color:#04d;background-color:#fff0f0"></span>            --right_fq reads_2.fastq <span style="color:#04d;background-color:#fff0f0">\
</span><span style="color:#04d;background-color:#fff0f0"></span>            --genome_lib_dir /path/to/ctat_genome_lib_dir <span style="color:#04d;background-color:#fff0f0">\
</span><span style="color:#04d;background-color:#fff0f0"></span>            --viral_db_fasta viruses.fasta
</code></pre></div><p>但其中主要是这个genome_lib_dir的文件需要构建参考库、当然也可以直接从已经构建上的<a href="https://data.broadinstitute.org/Trinity/CTAT_RESOURCE_LIB/">数据库</a>上下载。但是已经构建好的索引库太大，所以最好是下载其中源文件，然后自己进行构建库。具体可以看其他人的<a href="https://cloud.tencent.com/developer/article/1625294">文章</a>。</p>
<p>好了，当构建好索引库，也不是万事大吉。首先因为该软件的Github仓库中有子模块，所以基本上你下载下来后，还是会发现少了一些文件。</p>
<p>解决方法看知乎上<a href="https://zhuanlan.zhihu.com/p/361136073">文章</a>。 然后就算你用他的方法下载后，还是会发现<code>CTAT-VirusIntegrationFinder/util/bamsifter</code>中的htslib文件夹还有一个隐藏的子模块，所以上述操作还要再来一遍。下载完成后，你看在htslib文件中发现有个<code>makefile</code>的文件，终端输入<code>make</code>进行编译。</p>
<p>完成后，进行运行<code>ctat-VIF.py</code> ，然后就会报错，需要安装相应的python模块和R包，这个算是自己检查吧，但是接下来的才是一步步更加崩溃。</p>
<p>首先报错<code>libstdc++.so.6: version GLIBCXX_3.4.21' not found</code>，得了，需要进行LD库的更新，可以看<a href="https://blog.csdn.net/CV_Jason/article/details/84034220">文章</a>.</p>
<p>我是去更新了系统变量</p>
<pre tabindex="0"><code># gcc
export PATH=/share/app/gcc/9.3.0/bin:$PATH
export LD_LIBRARY_PATH=&quot;/share/app/gcc/9.3.0/lib64:$LD_LIBRARY_PATH&quot;
</code></pre><p>这样才解决问题。</p>
<p>然后又报错，我检查了原始作者的py脚本，最后发现是我的samtool版本太低了，所以报错了。我简直要气爆炸了。忍忍，继续更新samtools。</p>
<p>更新好就可以成功运行了</p></article>
    <footer class="post-footer">
      
      <ul class="post-tags">
        
          <li><a href="https://abego.cn/tags/software-install"><span class="tag">Software Install</span></a></li>
        
          <li><a href="https://abego.cn/tags/linux"><span class="tag">Linux</span></a></li>
        
          <li><a href="https://abego.cn/tags/python"><span class="tag">Python</span></a></li>
        
      </ul>
      
      <p class="post-copyright">
      <p>
        <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/deed.zh">LICENSE: CC-BY-NC 4.0协议 
        <i class="fab fa-creative-commons fa-2x"></i>
        <i class="fab fa-creative-commons-by fa-2x">
        </i><i class="fab fa-creative-commons-sa fa-2x"></i></a>
      </p>
        © 本站内容受CC-BY-NC 4.0协议保护,著作权归作者所有。商业转载请联系作者获得授权，非商业转载请注明出处。这篇文章创建在 <strong>2022-01-16</strong> , 内容的正确性和时效性请自行评估.
      </p>
    </footer>
    
      <div id="disqus_thread"></div>
<script type="application/javascript">
    var disqus_config = function () {
    
    
    
    };
    (function() {
        if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
            document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
            return;
        }
        var d = document, s = d.createElement('script'); s.async = true;
        s.src = '//' + "abego-cn" + '.disqus.com/embed.js';
        s.setAttribute('data-timestamp', +new Date());
        (d.head || d.body).appendChild(s);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
      
    
  </section>
  
<footer class="site-footer">
  <p>© 2019-2022 Woney&#39;s Blog</p>
  Powered by , the <a href="https://cran.r-project.org/" target="_blank" rel="noopener"><i class="fab fa-r-project"></i> </a><a href="https://github.com/rstudio/blogdown" target="_blank" rel="noopener">blogdown</a> package, and the 
    <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.</a> with theme <a href="https://github.com/laozhu/hugo-nuo" target="_blank" rel="noopener">Nuo</a>.</p>
  
</footer>


<script src="https://cdn.jsdelivr.net/npm/smooth-scroll@15.0.0/dist/smooth-scroll.min.js"></script>



<script async src="https://cdn.jsdelivr.net/npm/video.js@7.3.0/dist/video.min.js"></script>




<script async src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    tex2jax: {
      inlineMath: [['$','$'], ['\\(','\\)']],
      displayMath: [['$$','$$'], ['\\[','\\]']],
      processEscapes: true,
      processEnvironments: true,
      skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
      TeX: { equationNumbers: { autoNumber: "AMS" },
      extensions: ["AMSmath.js", "AMSsymbols.js"] }
    },
  });
</script>
<script type="text/x-mathjax-config">
  // Fix <code> tags after MathJax finishes running. This is a
  // hack to overcome a shortcoming of Markdown. Discussion at
  // https://github.com/mojombo/jekyll/issues/199
  MathJax.Hub.Queue(() => {
    MathJax.Hub.getAllJax().map(v => v.SourceElement().parentNode.className += ' has-jax');
  });
</script>



<script src="https://abego.cn/scripts/index.min.js"></script>

<script>
  if ('serviceWorker' in navigator) {
    navigator.serviceWorker.register('\/service-worker.js').then(function() {
      console.log('[ServiceWorker] Registered');
    });
  }
</script>




<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
	ga('create', 'UA-147459974-1', 'auto');
	
	ga('send', 'pageview');
}
</script>
<script async src='https://www.google-analytics.com/analytics.js'></script>






  </body>
</html>
