<!DOCTYPE html>
<html lang="zh-Hans">
<head>

  <meta charset="utf-8" />

  
  <title>BSseq-WDL：基于WDl的甲基化分析流程搭建和心得</title>

  
  
  <link rel="stylesheet"
        href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.4.0/styles/base16/dracula.min.css">
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.4.0/languages/python.min.js"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.4.0/languages/r.min.js"></script>
  <script>hljs.highlightAll();</script>

  
  <link href="//cdn.jsdelivr.net" rel="dns-prefetch">
  <link href="//cdnjs.cloudflare.com" rel="dns-prefetch">
  
  <link href="//use.fontawesome.com" rel="dns-prefetch">
  
  <link href="//fonts.googleapis.com" rel="dns-prefetch">
  <link href="//fonts.gstatic.com" rel="dns-prefetch">
  <link href="///disqus.com" rel="dns-prefetch">
  <link href="//c.disquscdn.com" rel="dns-prefetch">
  
  <link href="//www.google-analytics.com" rel="dns-prefetch">
  

  

  
  <meta name="author" content="Wang Jiaxuan">
  <meta name="description" content="做整合分析的时候，看到broad 研究所大佬写的软件太好，看了很久源码，学习wdl，小有所成，重构下之前用shell搭建的甲基化流程。
">

  
  
    <meta name="twitter:card" content="summary_large_image">
    <meta name="twitter:site" content="@abego13">
    <meta name="twitter:title" content="BSseq-WDL：基于WDl的甲基化分析流程搭建和心得">
    <meta name="twitter:description" content="做整合分析的时候，看到broad 研究所大佬写的软件太好，看了很久源码，学习wdl，小有所成，重构下之前用shell搭建的甲基化流程。
">
    <meta name="twitter:image" content="https://abego.cn/media/cover.jpg">
  

  
  <meta property="og:type" content="article">
  <meta property="og:title" content="BSseq-WDL：基于WDl的甲基化分析流程搭建和心得">
  <meta property="og:description" content="做整合分析的时候，看到broad 研究所大佬写的软件太好，看了很久源码，学习wdl，小有所成，重构下之前用shell搭建的甲基化流程。
">
  <meta property="og:url" content="https://abego.cn/2022/02/16/create-a-workflow-for-bs-seq-analysis-base-on-wdl/">
  <meta property="og:image" content="https://abego.cn/media/cover.jpg">




<meta name="generator" content="Hugo 0.91.2">


<link rel="canonical" href="https://abego.cn/2022/02/16/create-a-workflow-for-bs-seq-analysis-base-on-wdl/">

<meta name="renderer" content="webkit">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="format-detection" content="telephone=no,email=no,adress=no">
<meta http-equiv="Cache-Control" content="no-transform">


<meta name="robots" content="index,follow">
<meta name="referrer" content="origin-when-cross-origin">







<meta name="theme-color" content="#02b875">
<meta name="apple-mobile-web-app-capable" content="yes">
<meta name="apple-mobile-web-app-status-bar-style" content="black">
<meta name="apple-mobile-web-app-title" content="Woney&#39;s Blog">
<meta name="msapplication-tooltip" content="Woney&#39;s Blog">
<meta name='msapplication-navbutton-color' content="#02b875">
<meta name="msapplication-TileColor" content="#02b875">
<meta name="msapplication-TileImage" content="/icons/icon-144x144.png">
<link rel="icon" href="https://abego.cn/favicon.ico">
<link rel="icon" type="image/png" sizes="16x16" href="https://abego.cn/icons/icon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="https://abego.cn/icons/icon-32x32.png">
<link rel="icon" sizes="192x192" href="https://abego.cn/icons/icon-192x192.png">
<link rel="apple-touch-icon" href="https://abego.cn/icons/icon-152x152.png">
<link rel="manifest" href="https://abego.cn/manifest.json">


<link rel="preload" href="https://abego.cn/styles/main-rendered.min.css" as="style">


<link rel="preload" href="https://fonts.googleapis.com/css?family=Lobster" as="style">
<link rel="preload" href="https://abego.cn/images/avatar.png" as="image">
<link rel="preload" href="https://abego.cn/images/grey-prism.svg" as="image">


<style>
  body {
    background: rgb(244, 243, 241) url('/images/grey-prism.svg') repeat fixed;
  }
</style>
<link rel="stylesheet" href="https://abego.cn/styles/main-rendered.min.css">

<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.1/css/all.css" ZgotmplZ crossorigin="anonymous">


<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lobster">



<script src="https://cdn.jsdelivr.net/npm/medium-zoom@1.0.2/dist/medium-zoom.min.js"></script>





<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/video.js@7.3.0/dist/video-js.min.css">



  
  
<!--[if lte IE 8]>
  <script src="https://cdn.jsdelivr.net/npm/html5shiv@3.7.3/dist/html5shiv.min.js"></script>
  <script src="https://cdn.jsdelivr.net/npm/respond.js@1.4.2/dest/respond.min.js"></script>
  <script src="https://cdn.jsdelivr.net/npm/videojs-ie8@1.1.2/dist/videojs-ie8.min.js"></script>
<![endif]-->

<!--[if lte IE 9]>
  <script src="https://cdn.jsdelivr.net/npm/eligrey-classlist-js-polyfill@1.2.20180112/classList.min.js"></script>
<![endif]-->


</head>
  <body>
    <div class="suspension">
      <a role="button" aria-label="Go to top" title="Go to top" class="to-top is-hide"><span class="fas fa-caret-up" aria-hidden="true"></span></a>
      
        
	<a role="button" aria-label="Go to comments" title="Go to comments" class="to-comment" href="#disqus_thread"><span class="fas fa-comment-dots" aria-hidden="true"></span></a>
        
      
    </div>
    
    
  <header class="site-header">
  <a href="https://abego.cn/"><img class="avatar" src="https://abego.cn/images/avatar.png" alt="Avatar"></a>
  
  <h2 class="title"><a href="https://abego.cn/">Woney&#39;s Blog</a></h2>
  
  <p class="subtitle">生活 学习 工作</p>
  <button class="menu-toggle" type="button" aria-label="Main Menu" aria-expanded="false" tab-index="0">
    <span><i class="fas fa-bars"></i></span>
  </button>
  <nav class="site-menu collapsed">
    <h2 class="offscreen">Main Menu</h2>
    <ul class="menu-list">
      
      
      
      
        <li class="menu-item
          
          
          ">
          <a href="https://abego.cn/tags/">查找</a>
        </li>
      
        <li class="menu-item
          
          
           is-active">
          <a href="https://abego.cn/">主页</a>
        </li>
      
        <li class="menu-item
          
          
          ">
          <a href="https://abego.cn/resume/">简历</a>
        </li>
      
        <li class="menu-item
          
          
          ">
          <a href="https://abego.cn/links/">Links</a>
        </li>
      
        <li class="menu-item
          
          
          ">
          <a href="https://github.com/wangjiaxuan666">代码</a>
        </li>
      
        <li class="menu-item
          
          
          ">
          <a href="https://abego.cn/about/">吐槽</a>
        </li>
      
    </ul>
  </nav>
  <nav class="social-menu collapsed">
    <h2 class="offscreen">Social Networks</h2>
    <ul class="social-list"><li class="social-item">
          <a href="mailto:992914078@qq.com" title="Email" aria-label="Email">
            <span class="fas fa-envelope" aria-hidden="true"></span>
          </a>
        </li><li class="social-item">
          <a href="//github.com/wangjiaxuan666" rel="me" title="GitHub" aria-label="GitHub">
      <span class="fab fa-github" aria-hidden="true"></span>
          </a>
        </li><li class="social-item">
          <a href="//twitter.com/abego13" rel="me" title="Twitter" aria-label="Twitter">
            <span class="fab fa-twitter" aria-hidden="true"></span>
          </a>
        </li><li class="social-item">
          <a href="https://abego.cn/images/qrcode.jpg" rel="me" title="Wechat" aria-label="Wechat">
            <span class="fab fa-weixin" aria-hidden="true"></span>
          </a>
        </li></ul>
  </nav>
</header>

  <section class="main post-detail">
    <header class="post-header">
      <h1 class="post-title">BSseq-WDL：基于WDl的甲基化分析流程搭建和心得</h1>
      <p class="post-meta">@Wang Jiaxuan · Feb 16, 2022 · 6 min read</p>
    </header>
    
    <article class="post-content"><p>做整合分析的时候，看到broad 研究所大佬写的软件太好，看了很久源码，学习wdl，小有所成，重构下之前用shell搭建的甲基化流程。</p>
<h1 id="bsseq-wdl流程介绍">BSseq-WDl流程介绍</h1>
<h2 id="背景">背景</h2>
<p>BSseq-WDl 是我根据 BS-seeker2 + CGmapTools 为基础搭建的甲基化分析主要流程。说是主要，其实就是输出到甲基化结果文件（甲基化位点和甲基化率）。输出文件</p>
<blockquote>
<p>可以类比为RNA-seq输出了表达矩阵，但是后续的差异，可视化这些我都没开始弄。原因主要是工作是做单细胞和空间组的。而且表观基因组学一直是我认为组学里面最难的。所以如果想做的完善，肯定要投入很多时间和精力，我只是业余时间搞的，后续有时间再搞。但是输出表达矩阵应该对很多分析是够用的。</p>
</blockquote>
<h2 id="该流程的优点">该流程的优点</h2>
<p>说这个流程的优点，其实就是说wdl的优点，相比于之前用shell写的。我觉得wdl最大优点就是：</p>
<ul>
<li>彻底不用管文件的归档和存放，文件的输出和输入实现流程化控制</li>
<li>任务的并行运行，以及多个任务的多次调用比较方便</li>
<li>搭建虽然辛苦，但是维护起来比较简单</li>
</ul>
<p>但是说完优点，也要说缺点，真的也是有点难以忍受：</p>
<ul>
<li>没有积极活跃的社区，用户少，意味着问题很大概率要自己debug，让人崩溃</li>
<li>workflow，call，task 三者都要一个input，再加上一个output。一直让人来回输入那几个变量（因为要变量承上启下，还要声明），繁琐且容易报错，让人没有感觉到一点代码的简洁之美。</li>
<li>文件放在特定的目录下，层级太多，很难找到，必须搭配输出的json文件</li>
<li>if 语句中竟然还要评估否条件下的表达式，离谱！</li>
</ul>
<p>其他的缺点暂时想不到了，如果一句话概括就是：</p>
<blockquote>
<p>过程很痛苦，结果很美好，可能是流程控制语言的通病吧</p>
</blockquote>
<h2 id="流程空间流程">流程空间流程</h2>
<p>用wdl编写流程，需要cromwell.jar才能运行，输入文件有两个：</p>
<ul>
<li>参数输入文件：*.json</li>
<li>样本名称和fq路径表格文件：*.tsv</li>
</ul>
<p>输出文件需要添加 -m 参数，详情请见<strong>运行</strong>部分，输出的json文件：
包括了：</p>
<ul>
<li>过滤后的read</li>
<li>mapping后的用于call methylation的bam文件</li>
<li>甲基化结果文件<code>*.wig</code>, <code>*.ATCGmap</code>,<code>*.CGmap</code></li>
<li>call snp的结果文件<code>*.snv</code>,<code>*.vcf</code></li>
</ul>
<p>其中<code>*.ATCGmap</code>,<code>*.CGmap</code>是主要输出结果，前者代表是多种类型的甲基化位点结果（CHH，CHG，CG），后者只有CG类型的甲基化位点信息。结果是表格形式的，分别存储有位点，绝对甲基化reads数和相对的甲基化率等数值。</p>
<p>整体流程采用并行的方法，最大效率提高了运行速度和准确性：</p>
<p><img src="https://pic-1259340288.cos.ap-guangzhou.myqcloud.com/img/202202150912001644887520Ovne4Bimage-20220215091200415.png" alt="image-20220215091200415"></p>
<h2 id="环境要求">环境要求</h2>
<p>本流程运行需要以下环境：</p>
<ul>
<li>需要python 2.7（pysam）</li>
<li>需要java 1.8.0 的环境</li>
<li>环境中要有samtools，fastp，<a href="https://cgmaptools.github.io">CGmapTools</a>环境变量</li>
<li>需要下载<a href="https://github.com/BSSeeker/BSseeker2">BSseeker2</a></li>
</ul>
<p>关于BSseeker和CGmapTools的安装示例问题，可以参考后文</p>
<h2 id="流程运行">流程运行</h2>
<h3 id="软件下载">软件下载</h3>
<p>下载Github仓库，解压，在联网环境下，运行<code>download_cromwell.sh</code>脚本，下载<code>cromwell-58.jar</code>,如果已经有<code>cromwell</code>其他版本的可以不用下载，可以自行尝试是否版本兼容。</p>
<div class="highlight"><pre tabindex="0" style="background-color:#fff;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-bash" data-lang="bash">bash download_cromwell.sh
</code></pre></div><h3 id="配制文件">配制文件</h3>
<p>输入文件有两个，一个控制参数的<code>json</code>文件，一个是记录样本名称和fq序列文件路径的<code>tsv</code>表格.</p>
<h3 id="tsv-表格">tsv 表格</h3>
<p>制表符分割的文件，有三列，分别是样本名称，read1的路径，read2的路径。</p>
<p>Example:</p>
<pre tabindex="0"><code class="language-tsv" data-lang="tsv">sample_A	./test/sample_A.test.1k.R1.fq.gz	./test/sample_A.test.1k.R2.fq.gz
sample_B	./test/sample_B.test.1k.R1.fq.gz	./test/sample_B.test.1k.R2.fq.gz
</code></pre><blockquote>
<p>暂不考虑单端测序和生物学重复，前者是用不到，后者是还没做差异</p>
</blockquote>
<h3 id="json文件">json文件</h3>
<p>需要一些参数才能运行：</p>
<div class="highlight"><pre tabindex="0" style="background-color:#fff;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-json" data-lang="json">{
    <span style="color:#b06;font-weight:bold">&#34;bsseq.genome_index&#34;</span> : <span style="color:#d20;background-color:#fff0f0">&#34;error_path_just_for_test/test.fa_bowtie2/&#34;</span>,
    <span style="color:#b06;font-weight:bold">&#34;bsseq.python_file&#34;</span> : <span style="color:#d20;background-color:#fff0f0">&#34;wangjiaxuan/biosoft/miniconda3/envs/py27/bin/python&#34;</span>,
    <span style="color:#b06;font-weight:bold">&#34;bsseq.bsseeker_util_dir&#34;</span> : <span style="color:#d20;background-color:#fff0f0">&#34;wangjiaxuan/biosoft/BSseeker&#34;</span>,
    <span style="color:#b06;font-weight:bold">&#34;bsseq.refer_fa&#34;</span> : <span style="color:#d20;background-color:#fff0f0">&#34;./test/test.fa&#34;</span>,
    <span style="color:#b06;font-weight:bold">&#34;bsseq.alig_software&#34;</span> : <span style="color:#d20;background-color:#fff0f0">&#34;bowtie2&#34;</span>,
    <span style="color:#b06;font-weight:bold">&#34;bsseq.fastq_table&#34;</span> : <span style="color:#d20;background-color:#fff0f0">&#34;./test/input_sample_fq.tsv&#34;</span>,
    <span style="color:#b06;font-weight:bold">&#34;bsseq.step02_fq_qc&#34;</span> : <span style="color:#d20;background-color:#fff0f0">&#34;true&#34;</span>,
    <span style="color:#b06;font-weight:bold">&#34;bsseq.step03_bsseeker_align&#34;</span> : <span style="color:#d20;background-color:#fff0f0">&#34;true&#34;</span>
}
</code></pre></div><p>其中<code>genome_index</code>是基因组索引文件，这个是要BSseeker构建的。如果之前没有构建过，这个参数可以不写或者给了错误的地址。**流程都会自动根据参数<code>refer_fa</code>来进行基因组索引构建，只是需要的时间和内存都会比较大。</p>
<p>另外参数<code>refer_fa</code>是必须参数，除此以外，必须的参数还有python2.7路径的<code>python_file</code>，BSseeker的脚本目录<code>bsseeker_util_dir</code>，输入上文提到的tsv表格的<code>fastq_table</code>，以及比对软件<code>alig_software</code>。</p>
<h3 id="运行">运行</h3>
<h4 id="方法一">方法一</h4>
<p>运行流程, 我写了一个python的外包，运行<code>./bsseq-run -i bsseq.input.json</code>就可以开始分析，参数<code>-i</code>控制输入的json文件，<code>-c</code>是指定的cromwell.jar的路径，默认是在WDL文件下的。</p>
<div class="highlight"><pre tabindex="0" style="background-color:#fff;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-python" data-lang="python"><span style="color:#888"># 参数</span>
<span style="color:#888"># usage: bsseq-run [-h] --input INPUT [--cromwell CROMWELL] [--version]</span>

<span style="color:#888"># BS-seq analysis workflow base on bsseeker and CGmaptools</span>

<span style="color:#888"># optional arguments:</span>
<span style="color:#888">#  -h, --help            show this help message and exit</span>
<span style="color:#888">#  --input INPUT, -i INPUT</span>
<span style="color:#888">#                       the input json</span>
<span style="color:#888">#  --cromwell CROMWELL, -c CROMWELL</span>
<span style="color:#888">#                       Optional path to cromwell jar file</span>
<span style="color:#888">#  --version, -v         show version tag: 0.1</span>
</code></pre></div><h4 id="方法二">方法二</h4>
<p>当然外包的python又需要python3的环境，而运行BSseeker是python2的环境，我也没无奈，因为我不会python2，所以嫌麻烦也可以直接运行命令行：</p>
<div class="highlight"><pre tabindex="0" style="background-color:#fff;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-bash" data-lang="bash">java -jar WDL/WDL/cromwell-58.jar run WDL/bsseq.wdl -i bsseq.input.json -m bsseq.output.json
<span style="color:#888"># -m 参数是输出文件，可以不写，但是不写后续分析很麻烦</span>
</code></pre></div><h3 id="结果解读">结果解读</h3>
<p>结果可能有点疑惑，但这也是wdl的特性，输出文件都会在cromwell-executions文件里，里面根据task来进行分文件，又进一步细分为input和execution。但是看<code>bsseq.output.json</code>就不会疑惑了。</p>
<p>有用的信息，我都output了</p>
<p>Example：</p>
<div class="highlight"><pre tabindex="0" style="background-color:#fff;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-json" data-lang="json">  <span style="color:#d20;background-color:#fff0f0">&#34;outputs&#34;</span><span style="color:#a61717;background-color:#e3d2d2">:</span> {
    <span style="color:#b06;font-weight:bold">&#34;bsseq.bsseeker_call_methylation_bam_bai&#34;</span>: [<span style="color:#d20;background-color:#fff0f0">&#34;sort_rmSX.bam.bai&#34;</span>, <span style="color:#d20;background-color:#fff0f0">&#34;sort_rmSX.bam.bai&#34;</span>],
    <span style="color:#b06;font-weight:bold">&#34;bsseq.bsseeker_call_methylation_bam&#34;</span>: [<span style="color:#d20;background-color:#fff0f0">&#34;sort_rmSX.bam&#34;</span>],
    <span style="color:#b06;font-weight:bold">&#34;bsseq.sample&#34;</span>: [<span style="color:#d20;background-color:#fff0f0">&#34;sample_A&#34;</span>, <span style="color:#d20;background-color:#fff0f0">&#34;sample_B&#34;</span>],
    <span style="color:#b06;font-weight:bold">&#34;bsseq.fastp_filter_fq1&#34;</span>: [<span style="color:#d20;background-color:#fff0f0">&#34;sample_B.filter.R1.fq.gz&#34;</span>],
    <span style="color:#b06;font-weight:bold">&#34;bsseq.bsseeker_methy_CGmap&#34;</span>: [<span style="color:#d20;background-color:#fff0f0">&#34;merge.CGmap&#34;</span>],
    <span style="color:#b06;font-weight:bold">&#34;bsseq.bsseeker_index&#34;</span>: <span style="color:#d20;background-color:#fff0f0">&#34;test.fa_bowtie2&#34;</span>,
    <span style="color:#b06;font-weight:bold">&#34;bsseq.bsseeker_methy_wig&#34;</span>: [<span style="color:#d20;background-color:#fff0f0">&#34;merge.wig&#34;</span>],
    <span style="color:#b06;font-weight:bold">&#34;bsseq.cgmaptools_methy_snv&#34;</span>: [<span style="color:#d20;background-color:#fff0f0">&#34;bayes.snv&#34;</span>],
    <span style="color:#b06;font-weight:bold">&#34;bsseq.fastp_filter_fq2&#34;</span>: [<span style="color:#d20;background-color:#fff0f0">&#34;sample_B.filter.R2.fq.gz&#34;</span>],
    <span style="color:#b06;font-weight:bold">&#34;bsseq.cgmaptools_methy_vcf&#34;</span>: [<span style="color:#d20;background-color:#fff0f0">&#34;bayes.vcf&#34;</span>],
    <span style="color:#b06;font-weight:bold">&#34;bsseq.bsseeker_methy_ATCGmap&#34;</span>: [<span style="color:#d20;background-color:#fff0f0">&#34;merge.ATCGmap&#34;</span>]
    }
<span style="color:#a61717;background-color:#e3d2d2">#</span> <span style="color:#a61717;background-color:#e3d2d2">信息匿去部分信息，本身是按照样本输出的</span>
</code></pre></div><h3 id="甲基化结果文件展示">甲基化结果文件展示：</h3>
<p>我们拿ATCGmap文件举例，结果是文本格式，包含甲基化位点和甲基化水平 <img src="https://pic-1259340288.cos.ap-guangzhou.myqcloud.com/img/202202161753521645005232A6WlOximage-20220216175351916.png" alt="image-20220216175351916"></p>
<p>具体每一列的详细信息如下：</p>
<p><img src="https://pic-1259340288.cos.ap-guangzhou.myqcloud.com/img/202202161755271645005327t10sDximage-20220216175527691.png" alt="image-20220216175527691"></p>
<h2 id="安装">安装</h2>
<h3 id="安装bsseeker2">安装BSseeker2</h3>
<p>按照BSseeker的<a href="http://pellegrini-legacy.mcdb.ucla.edu/bs_seeker2/">网址</a>上的介绍安装和使用
简单点话就是</p>
<div class="highlight"><pre tabindex="0" style="background-color:#fff;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-bash" data-lang="bash">git clone https://github.com/BSSeeker/BSseeker2
<span style="color:#038">cd</span> BSseeker2
</code></pre></div><p>就可以看到其中的py脚本，但是亲测python 3.8不能用，只能用conda创建一个python2.7的环境，同时需要环境中有bowtie, bowtie2, soap三个软件中任意一个都可以。python环境也要有<code>pysam</code>模块。</p>
<h3 id="安装cgmaptools">安装CGmapTools</h3>
<div class="highlight"><pre tabindex="0" style="background-color:#fff;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-bash" data-lang="bash">git clone https://github.com/guoweilong/cgmaptools
bash install.sh
<span style="color:#888"># add cgmaptools to your system PATH if necessary</span>
<span style="color:#888"># open ~/.bashrc</span>
vi ~/.bashrc
<span style="color:#888"># add the following lines to the end of ~/.bashrc</span>
<span style="color:#038">export</span> <span style="color:#369">PATH</span>=/path/to/cgmaptools/:<span style="color:#369">$PATH</span>
<span style="color:#888"># then, souce your ~/.bashrc</span>
<span style="color:#038">source</span> ~/.bashrc
</code></pre></div><p>还需要在python2.7的环境中安装pysam和scipy</p>
<div class="highlight"><pre tabindex="0" style="background-color:#fff;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-bash" data-lang="bash">conda activate py27
pip install pysam
pip install scipy
</code></pre></div><h2 id="资料来源">资料来源</h2>
<p>资料主要来源黄湘仪的教程——<a href="https://guoweilong.github.io/BSseeker2_CGmapTools_tutorial_181014.pdf">18.10.04亚硫酸盐测序数据基本分析流程代码及软件</a>，当然也感谢BSseeker2 和CGmapTools开发者农大的<a href="http://guoweilong.github.io/">郭伟龙</a>博士（中国农业大学）和朱平博士（天津血研所/北京大学）等人合作开发了DNA甲基化数据分析工具包：<a href="https://cgmaptools.github.io/">CGmapTools</a>，包含40个独立的命令行工具，提供了全面的DNA甲基化数据分析和可视化功能。目前该工作已经在生物信息学著名期刊<strong>Bioinformatics</strong>上以original paper形式发表。</p>
<blockquote>
<p>Weilong Guo <em>, Ping Zhu</em> , et al. (2017), <a href="https://doi.org/10.1093/bioinformatics/btx595">CGmapTools improves the precision of heterozygous SNV calls and supports allele-specific methylation detection and visualization in bisulfite-sequencing data</a>.</p>
</blockquote>
<p><strong>再次感谢各位大佬创建这么好用的软件</strong></p></article>
    <footer class="post-footer">
      
      <ul class="post-tags">
        
          <li><a href="https://abego.cn/tags/software-install"><span class="tag">Software Install</span></a></li>
        
          <li><a href="https://abego.cn/tags/%E6%B5%81%E7%A8%8B"><span class="tag">流程</span></a></li>
        
      </ul>
      
      <p class="post-copyright">
      <p>
        <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/deed.zh">LICENSE: CC-BY-NC 4.0协议 
        <i class="fab fa-creative-commons fa-2x"></i>
        <i class="fab fa-creative-commons-by fa-2x">
        </i><i class="fab fa-creative-commons-sa fa-2x"></i></a>
      </p>
        © 本站内容受CC-BY-NC 4.0协议保护,著作权归作者所有。商业转载请联系作者获得授权，非商业转载请注明出处。这篇文章创建在 <strong>2022-02-16</strong> , 内容的正确性和时效性请自行评估.
      </p>
    </footer>
    
      <div id="disqus_thread"></div>
<script type="application/javascript">
    var disqus_config = function () {
    
    
    
    };
    (function() {
        if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
            document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
            return;
        }
        var d = document, s = d.createElement('script'); s.async = true;
        s.src = '//' + "abego-cn" + '.disqus.com/embed.js';
        s.setAttribute('data-timestamp', +new Date());
        (d.head || d.body).appendChild(s);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
      
    
  </section>
  
<footer class="site-footer">
  <p>© 2019-2022 Woney&#39;s Blog</p>
  Powered by , the <a href="https://cran.r-project.org/" target="_blank" rel="noopener"><i class="fab fa-r-project"></i> </a><a href="https://github.com/rstudio/blogdown" target="_blank" rel="noopener">blogdown</a> package, and the 
    <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.</a> with theme <a href="https://github.com/laozhu/hugo-nuo" target="_blank" rel="noopener">Nuo</a>.</p>
  
</footer>


<script src="https://cdn.jsdelivr.net/npm/smooth-scroll@15.0.0/dist/smooth-scroll.min.js"></script>



<script async src="https://cdn.jsdelivr.net/npm/video.js@7.3.0/dist/video.min.js"></script>




<script async src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    tex2jax: {
      inlineMath: [['$','$'], ['\\(','\\)']],
      displayMath: [['$$','$$'], ['\\[','\\]']],
      processEscapes: true,
      processEnvironments: true,
      skipTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
      TeX: { equationNumbers: { autoNumber: "AMS" },
      extensions: ["AMSmath.js", "AMSsymbols.js"] }
    },
  });
</script>
<script type="text/x-mathjax-config">
  // Fix <code> tags after MathJax finishes running. This is a
  // hack to overcome a shortcoming of Markdown. Discussion at
  // https://github.com/mojombo/jekyll/issues/199
  MathJax.Hub.Queue(() => {
    MathJax.Hub.getAllJax().map(v => v.SourceElement().parentNode.className += ' has-jax');
  });
</script>



<script src="https://abego.cn/scripts/index.min.js"></script>

<script>
  if ('serviceWorker' in navigator) {
    navigator.serviceWorker.register('\/service-worker.js').then(function() {
      console.log('[ServiceWorker] Registered');
    });
  }
</script>




<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
	ga('create', 'UA-147459974-1', 'auto');
	
	ga('send', 'pageview');
}
</script>
<script async src='https://www.google-analytics.com/analytics.js'></script>






  </body>
</html>
